### PR TITLE
refactor: remove unused methods/types in `src/sugar`

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

Title says it all.

### Context of Change

xrpl.js cleanup

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)

## Test Plan

CI passes.

### CODE REVIEW

**Comment by mvadari** [comment]
waiting on #1622

File: src/sugar/utils.ts [hunk 1]
```diff
@@ -1,125 +1,27 @@
...
+ * @returns The account's classic address.
+ * @throws Error if the X-Address has an associated tag.
+ */
+export function ensureClassicAddress(account: string): string {
+  if (isValidXAddress(account)) {
+    const { classicAddress, tag } = xAddressToClassicAddress(account)
+
+    // Except for special cases, X-addresses used for requests
+    // must not have an embedded tag. In other words,
+    // `tag` should be `false`.
```
Comment (JST5000) [inline src/sugar/utils.ts:16]: Do those special cases use this function, or do they handle it separately?

**Review by JST5000 — APPROVED** [review]
LGTM (One question, but it's for my curiousity)

File: src/sugar/utils.ts (hunk 1 above)
Comment (mvadari) [reply src/sugar/utils.ts:16]: Pretty sure that's talking about transactions, which do handle X-Addresses separately (see `autofill`).

Comment (natenichols) [reply src/sugar/utils.ts:16]: Should this function go to `./src/utils` or something. isn't it only uses in `Client.request`?

File: src/sugar/index.ts [hunk 2]
```diff
@@ -1,33 +0,0 @@
-import { xAddressToClassicAddress, isValidXAddress } from 'ripple-address-codec'
```
Comment (natenichols) [inline src/sugar/index.ts:1]: Why did we delete `sugar/index.ts` How are we exporting from sugar?

**Review by natenichols — APPROVED** [review]
One question, but its not directly relevant to the PR, so I'm okay w/ giving approval

**Pull request merged**